Composer/packages/server/src/services/project.ts

```diff
@@ -204,7 +204,11 @@ export class BotProjectService {
     }
   };
 
-  public static openProject = async (locationRef: LocationRef, user?: UserIdentity): Promise<string> => {
+  public static openProject = async (
+    locationRef: LocationRef,
+    user?: UserIdentity,
+    allowPartialBots?: boolean
```

Comment:
I'm mixed on this new interface. This is a critical path module and I'm wondering if we can preserve the interface.

options:

- change interface to accept `options` instead of `allowPartialBots` declared initially with `{ allowPartialBots: boolean }` so that it is future-proofed a bit.
- try/catch the error at the call-site and check the message. if we're migrating pass, re-throw if not
- something else?

Reply:
what this flag allows is for it to *not* throw if we are migrating -- otherwise, it will reject the project as a non-bot because it doesn't YET include the necessary files.

I will go with option 1 here -- adding an options param instead of the boolean.